hub/core/storage/gcs.py

```diff
@@ -277,6 +277,23 @@ def clear(self):
             except Exception:
                 pass
 
+    def rename(self, root):
+        """Rename root folder"""
+        self.check_readonly()
+        path = root.replace("gcs://", "").replace("gcp://", "")
+        new_bucket, new_path = path.split("/", 1)
+        if new_bucket != self.client_bucket.name:
+            raise RenameError
+        blob_objects = self.client_bucket.list_blobs(prefix=self.path)
+        for blob in blob_objects:
```

Comment:
this is happening serially, can't we utilize the multi-worker copy that you had built? Or is there some difference?